src/views/search/components/AdvancedSearchModal/components/ProjectMultiSelect.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FC, useMemo } from 'react';
+
+import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import useProjects from '@kubevirt-utils/hooks/useProjects';
+import { MultiTypeaheadSelect, MultiTypeaheadSelectOption } from '@patternfly/react-templates';
+
+type ProjectMultiSelectProps = {
+  projects: string[];
+  setProjects: React.Dispatch<React.SetStateAction<string[]>>;
+};
+
+const ProjectMultiSelect: FC<ProjectMultiSelectProps> = ({ projects, setProjects }) => {
+  const [projectNames] = useProjects();
+  const projectOptions = useMemo<MultiTypeaheadSelectOption[]>(
+    () =>
+      projectNames.map((projectName) => ({
+        content: projectName,
+        selected: projects.includes(projectName),
+        value: projectName,
+      })),
+    [projects, projectNames],
+  );
+
+  return (
+    <MultiTypeaheadSelect
+      onSelectionChange={(_e, selectedProjects: string[]) => {
+        setProjects(selectedProjects);
+      }}
+      initialOptions={projectOptions}
+      isScrollable
+      placeholder={projects.length ? t('Select project') : t('All projects')}
```

Comment:
!isEmpty(projects)

Reply:
`projects.length > 0` is better than `!isEmpty(projects)` because it conveys the fact that `projects` is an array, while `isEmpty` is a utility function that works with both arrays and objects so that adds unnecessary complexity.

We really just need to check if the array is not empty.